modules/reindex/src/main/java/org/elasticsearch/index/reindex/DeleteByQueryRequest.java

```diff
@@ -76,4 +81,10 @@ public String toString() {
         searchToString(b);
         return b.toString();
     }
+
+    @Override
+    public List<? extends IndicesRequest> subRequests() {
+        assert getSearchRequest() != null;
+        return Collections.unmodifiableList(Collections.singletonList(getSearchRequest()));
```

Comment:
I think what gets returned by singletonList is already immutable, no need to wrap it with unmodifiableList too.